### PR TITLE
Support for matrix data init. types

### PR DESCRIPTION
**Pull request by jkottiku**
- Support for new random integer and trignometric float values added in rvs_blas.
- New parameter matrix init. type added in gst conf. file with trig and rand values.
- Updated MI300X gst_single.conf with both trignometric float and random interger tests for fp8, fp16, bf16, sgemm and dgemm.
- Removed decimal in GFLOPS from the output logs.
- Other minor logging changes.

### CODE REVIEW

File: gst.so/include/action.h [hunk 1]
```diff
@@ -87,6 +87,9 @@ class gst_action: public rvs::actionbase {
     uint64_t gst_matrix_size_b;
     uint64_t gst_matrix_size_c;
 
+    //! matrix initialization
```
Comment (manoj-freyr) [inline gst.so/include/action.h:90]: state in comments what it actually does

File: rvs/conf/MI300X/gst_single.conf [hunk 2]
```diff
@@ -24,94 +24,196 @@
...
   ramp_interval: 5000
-  duration: 30000
+  duration: 15000
   copy_matrix: false
-  target_stress: 1000000
+  target_stress: 900000
   matrix_size_a: 8192
   matrix_size_b: 8192
   matrix_size_c: 8192
+  matrix_init: trig
```
Comment (manoj-freyr) [inline rvs/conf/MI300X/gst_single.conf:38]: mention the usage detail at top of the conf file as a comment

File: src/rvs_blas.cpp [hunk 3]
```diff
@@ -26,11 +26,48 @@
...
+#include <cmath>
+#include <random>
+#include <thread>
+
+/* ============================================================================================ */
+// Random number generator
+using rvsblas_rng_t = std::mt19937;
+
+// Random number generator
+rvsblas_rng_t rvsblas_seed(69069); // A fixed seed to start at
```
Comment (manoj-freyr) [inline src/rvs_blas.cpp:38]: do we need not randomise this initial seed?

File: src/rvs_blas.cpp [hunk 4]
```diff
@@ -758,91 +797,129 @@ void rvs_blas::generate_random_matrix_data(void) {
...
  * @brief fast pseudo random generator 
  * @return floating point random number
  */
-float rvs_blas::fast_pseudo_rand(uint64_t *nextr) {
+float rvs_blas::fast_pseudo_rand(uint64_t *nextr, size_t i) {
+
+  if ("rand" == matrix_init) {
+
+    if("fp8_r" == data_type) {
+      return (float)std::uniform_int_distribution<int>(1, 2)(rvsblas_t_rng);
```
Comment (manoj-freyr) [inline src/rvs_blas.cpp:901]: why parametrizing as int? wont it be only integral values1 and 2 which we are converting to float? will it make more sense to parametrizing it as float?

File: src/rvs_blas.cpp [hunk 5]
```diff
@@ -758,91 +797,129 @@ void rvs_blas::generate_random_matrix_data(void) {
...
+    else if("bf16_r" == data_type)
+    {
+      return (float)std::uniform_int_distribution<int>(-2, 2)(rvsblas_t_rng);
+    }
+    else { /* sgemm, dgemm */
+      return rvsblas_uniform_int_1_10();
+    }
+  }
+  else if ("trig" == matrix_init) {
+    return static_cast<float>(sin(i));
```
Comment (manoj-freyr) [inline src/rvs_blas.cpp:916]: better to cast argument to float than to cast return type? looks like arg, if int, will be treated as double which returns double as result and we are downcasting it to float in this case?

**Review by manoj-freyr — COMMENTED** [review]
check the comments

File: gst.so/include/action.h (hunk 1 above)
Comment (jkottiku) [reply gst.so/include/action.h:90]: done

File: src/rvs_blas.cpp (hunk 3 above)
Comment (jkottiku) [reply src/rvs_blas.cpp:38]: This is just the main thread seed at init, other seeds are based on thread ids.

File: src/rvs_blas.cpp (hunk 4 above)
Comment (jkottiku) [reply src/rvs_blas.cpp:901]: Followed the same logic as in rocblas-bench.

File: src/rvs_blas.cpp (hunk 5 above)
Comment (jkottiku) [reply src/rvs_blas.cpp:916]: Done

File: rvs/conf/MI300X/gst_single.conf (hunk 2 above)
Comment (jkottiku) [reply rvs/conf/MI300X/gst_single.conf:55]: Done

**Review by manoj-freyr — APPROVED** [review]
looks good, attach the results in the ticket to see the difference in perf numbers

**Comment by jkottiku** [comment]
Test logs attached in the ticket.

**Pull request merged**